src/eval/threadListableOperands.cpp

```diff
@@ -59,9 +62,54 @@ ThreadListableOperandsReturnType threadListableOperands(const ast::FunctionCall&
 		resultOperands[i] = ast::Node::make<ast::FunctionCall>(function, resultListOperands);
 	}
 
+	if ( std::all_of(resultOperands.begin(), resultOperands.end(), isEvaluationOfNodeThreadSafe) ) {
+		__gnu_parallel::for_each(resultOperands.begin(), resultOperands.end(), [&](ast::Node& node) {
+			node = sessionEnvironment.recursiveEvaluate(node);
+		});
+	}
+
```

Comment:
Well I think if the hold attribute (or any of it's variants) is set for head, then recursiveEvaluate will nop for the nodes (albeit on many threads). But if you can give an example of a call where this isn't the case, I'll gladly add a unittest for it, and see if it fails.